bpf/lib/nat.h

```diff
@@ -312,24 +313,25 @@ snat_v4_nat_handle_mapping(struct __ctx_buff *ctx,
 	if (*state) {
 		int ret;
 		struct ipv4_ct_tuple rtuple = {};
-
-		if (target->addr == (*state)->to_saddr &&
+		/* reverse SNAT entries could be missing due to LRU eviction */
+		set_v4_rtuple(tuple, *state, &rtuple);
+		rstate = __snat_lookup(map, &rtuple);
+		if (rstate &&
+		    target->addr == (*state)->to_saddr &&
 		    needs_ct == (*state)->common.needs_ct) {
 			barrier_data(*state);
 			return 0;
```

Comment:
Aside: should this be replaced with `CTX_ACT_OK`?

Reply:
This is not the final stage of the packet. Returning CTX_ACT_OK may be too early?

Reply:
`CTX_ACT_OK` is just an alias for `0`

Don't worry about this though, marked it as an aside (i.e. a follow-up for anyone).

Reply:
Yup.  I was checking  the convention of  using `CTX_ACT_OK`,  other places seem to be the final stages.